sharedbuffers/mapped_struct.py

```diff
@@ -297,6 +306,226 @@ def unpack_from(cls, buf, offs, idmap = None, array = array.array, itemsizes = {
             raise ValueError("Inconsistent data, unknown type code %r" % (dcode,))
         return rv
 
+@cython.cclass
+class proxied_list(object):
+    cython.declare(
+        buf = object,
+        pybuf = 'Py_buffer',
+        offs = cython.ulonglong,
+        dataoffs = cython.ulonglong,
+        itemsize = cython.ulonglong,
+        objlen = cython.ulonglong,
```

Comment:
I don't like all the state the proxy has.

For one, it makes them big. For another, it makes them sensitive to buffer changes. We don't want to paint ourselves into the read-only buffer corner, we will implement writable buffers eventually, so the proxy should only point to the list header, and things like the dcode, element size and all that should be read from the header at item access time. If properly cythonized, the overhead shouldn't be big.